style/hudl_sql_style_guide.md

```diff
@@ -5,33 +5,101 @@ description: A guide to writing clean, clear, and consistent SQL.
+
+  __Business-centric__ models should be created within the marts directory. __Source-centric__ models should be created within the staging directory. 
+
+> Many pre-existing business-centric models do not exist within the marts directory.
+            However, all new ones should be created there.
+
+* Define raw data as dbt sources: No raw data sources should be referenced other than in sources. If a source and staging layer do not exist for a model you want to build, you are required to build the needed sources and staging layers.
+* __Staging__ 
+  - Used to clean sources
+  - Rename and Recast fields in this layer
+  - There should only be one staging file per one source. Only seed files should be joined at this layer if necessary.
+  - No business metrics/values should be created at this layer
+
+* __dbt seed__
+  - All seed files belong in the `data` directory.
+  - Seed files are best used for mappings (ie. country codes to country names)
+* __dbt snapshots__
+  - Use to track slowly changing dimensions
+  - As a general rule should be built on top of sources
+* __dbt macros__
+  - Best for trusted, reusable pieces of logic (ie.fiscal year creation).
+
+### File Materialization
+* __Table__
+  - Use the table materialization for any models being queried by BI tools (ie. Looker), to give your end user a faster experience
+  - Also use the table materialization for any slower transformations that are used by many downstream models
+  - Are fast to query but take a long time to build, especially when they contain complex transformations and or a large amount of data, so keep this in mind when     selecting this materialization
+* __View__
+  - Builds faster, but requires more time for the end user to query. 
+  - Is dbt’s default materialization if you do not specify one
+  - Generally start with views for your models, and only change to another materialization when you're noticing performance problems.
+  - Views are best suited for models that do not do significant transformation, e.g. renaming, recasting columns.
+  - As a general rule, all sources should be materialized as views
+  - As a general rule, you will want the view to be late-binding. This is currently the default in our dbt project and does not need to be specified in every file.     For  more information on why we do this, you can check out this article: https://blog.getdbt.com/using-redshift-s-late-binding-views-with-dbt/
+
+* __Ephemeral__
+  - Use for very light-weight transformations that are early on in your DAG
+  - Use when code will be re-used in multiple files to help keep code DRY (do not repeat yourself). However, if this code won’t be re-used, consider making it a       CTE rather than it’s own file.
```

Comment:
This is defined above
```suggestion
  - Use when code will be re-used in multiple files to help keep code DRY. However, if this code won’t be re-used, consider making it a       CTE rather than it’s own file.
```